src/Data/BEncode/Parser.hs

```diff
@@ -82,12 +82,11 @@ dict name = BParser $ \b -> case b of
                               BDict _ -> Error $ "Name not found in dictionary: " ++ name
                               _ -> Error $ "Not a dictionary: " ++ name
 
-list :: String -> BParser a -> BParser [a]
-list name p
-    = dict name >>= \lst ->
-      BParser $ \b -> case lst of
-                      BList bs -> foldr (cat . runB p) (Ok [] b) bs
-                      _ -> Error $ "Not a list: " ++ name
+list :: BParser a -> BParser [a]
+list p
+    = BParser $ \lst -> case lst of
+          BList (bs) -> foldr (cat . runB p) (Ok [] (BList [])) bs
+          _ -> Error $ "Not a list: " ++ (show lst)
     where cat (Ok v _) (Ok vs b) = Ok (v:vs) b
```

Comment:
Can you explain? Is it the `(Ok [] (BList []))` parameter to `foldr` you are looking at? I was thinking of just making that parameter `(Ok [] undefined)`, thinking that the second param to the `Ok` constructor never would get evaluated. If it's possible to make it blow up, I'm sure some simple input in ghci should find a case, no?